src/solidOpts.ts

```diff
@@ -21,7 +24,24 @@ let elementDeleteQueue: ElementNode[] = [];
 function flushDeleteQueue(): void {
   for (let el of elementDeleteQueue) {
     if (Number(el._queueDelete) < 0) {
-      el.destroy();
+      if (el.preserve) {
```

Comment:
```suggestion
      if ("preserve" in el) {
```
In a sense when you pass `null` as the owner it should stay alive forever, and not default to being disposed when removed from parent.
Because `null` means "top-level owner". Like if you create an effect without a parent owner it will never dispose.
Which is another point to why `preserve: boolean` would be better.